tests/gtc/regression/cpp/CMakeLists.txt

```diff
@@ -32,25 +40,49 @@ function(add_regression_test_executable)
     cmake_parse_arguments(ARG "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
     message(STATUS "${ARG_NAME}")
-    set(generated_stencil_source ${CMAKE_CURRENT_SOURCE_DIR}/generated_${ARG_NAME}.hpp)
-    add_custom_command(OUTPUT ${generated_stencil_source}
+
+    # generate c++ from gtscript
+    set(STENCIL_IMPL_SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/generated_${ARG_NAME}.hpp)
+    add_custom_command(OUTPUT ${STENCIL_IMPL_SOURCE}
         COMMAND ${Python3_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/${ARG_INPUT} ${ARG_MODE}
         DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${ARG_INPUT})
-    add_custom_target(generated_${ARG_NAME} DEPENDS ${generated_stencil_source})
-    add_executable(${ARG_NAME} ${ARG_SOURCES}) # TODO fix GridTools appending .cu to .hpp, then we don't need the custom_target
+    add_custom_target(generated_${ARG_NAME} DEPENDS ${STENCIL_IMPL_SOURCE})
+
+
+    # generate driver code
+    #  create directory to generate code to
+    set(gen_dir ${CMAKE_CURRENT_BINARY_DIR}/generated_${ARG_NAME})
+    file(MAKE_DIRECTORY ${gen_dir})
+    #  the gridtools_setup_target expect a relative path so transform it
+    file(RELATIVE_PATH bin_dir_rel_to_src_dir ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/.)
+    set(generated_arg_sources)
+    foreach(arg_source ${ARG_SOURCES})
+        hash_path(${arg_source} generated_source)
```

Comment:
Why?

Reply:
If you have multiple regression tests with the same name but different path and just use the name you get a collision. Since this is hard to debug I tried to avoid it before any future confusion.

Reply:
Mh. Previously the hash included the target which was required to differentiate between gpu and cpu target, but with the targent dependent directory this is not needed anymore and having two files with the same filename, but different paths in the target is a very rare case. I'll remove it.